mirrord-layer/src/pod_api.rs

```diff
@@ -382,7 +388,13 @@ impl KubernetesAPI {
             .and_then(|pod| pod.metadata.name.clone())
             .ok_or(LayerError::JobPodNotFound(mirrord_agent_job_name))?;
 
-        wait_for_agent_startup(&pod_api, &pod_name, "mirrord-agent".to_string()).await?;
+        wait_for_agent_startup(
```

Comment:
It could be that the agent doesn't get to the running phase and it's stuck there. I'd wrap all with a timeout imo.